### PR TITLE
🌱 Refactor some code

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactors code. Minor behavior change - instead of `--raw` option, use `--format=raw`.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: cmd/package_managers.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	"fmt"
+	"net/http"
+	"time"
+
+	sce "github.com/ossf/scorecard/v4/errors"
+)
+
+var errInvalidInput = errors.New("only one package name must be non-empty")
+
+// TODO(cmd): Add unit tests for this.
```
Comment (naveensrinivasan) [inline cmd/package_managers.go:30]: Can you please create an issue for this?

File: cmd/package_managers.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+func fetchGitRepositoryFromPackageManagers(npm, pypi, rubygems string) (bool, string, error) {
```
Comment (naveensrinivasan) [inline cmd/package_managers.go:31]: IMO tuple is good to read in returns, anything more than `2` should be a struct. It makes it easier to maintain and read. This is with respect to  `bool, string, error`

File: cmd/root.go [hunk 3]
```diff
@@ -400,154 +387,43 @@ func getEnabledChecks(sp *spol.ScorecardPolicy, argsChecks []string,
...
-	err = json.NewDecoder(resp.Body).Decode(v)
-	if err != nil {
-		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to parse ruby gem json: %v", err))
-	}
-	if v.SourceCodeURI == "" {
-		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("could not find source repo for ruby gem: %v", err))
-	}
-	return v.SourceCodeURI, nil
+func getRepoAccessors(ctx context.Context, repoURI, localURI string, logger *sclog.Logger) (
+	clients.Repo, // repo
```
Comment (naveensrinivasan) [inline cmd/root.go:391]: I like the comments on these returns. Thanks 👍

**Review by naveensrinivasan — APPROVED** [review]
Thank you! Few suggestions on the code. But I am OK if you think these aren't required from your perspective. Feel free to merge 👍

File: cmd/package_managers.go (hunk 1 above)
Comment (azeemshaikh38) [reply cmd/package_managers.go:30]: Done - #1568

File: cmd/package_managers.go (hunk 2 above)
Comment (azeemshaikh38) [reply cmd/package_managers.go:31]: Done.

File: clients/localdir/repo.go [hunk 4]
```diff
@@ -78,10 +72,7 @@ func (r *repoLocal) AppendMetadata(m ...string) {
 
 // MakeLocalDirRepo returns an implementation of clients.Repo interface.
 func MakeLocalDirRepo(pathfn string) (clients.Repo, error) {
-	if !strings.HasPrefix(pathfn, filePrefix) {
-		return nil, fmt.Errorf("%w", errInvalidURI)
-	}
-	p := path.Clean(pathfn[len(filePrefix):])
```
Comment (laurentsimon) [inline clients/localdir/repo.go:84]: make sure to test nothin breaks after this change.

File: cmd/package_managers.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
```
Comment (laurentsimon) [inline cmd/package_managers.go:5]: is this something anyone uses in practice? Is it worth maintaining?

**Review by laurentsimon — APPROVED** [review]

File: clients/localdir/repo.go (hunk 4 above)
Comment (azeemshaikh38) [reply clients/localdir/repo.go:84]: Tested with `--local` on my workstation and it worked fine.

File: cmd/package_managers.go (hunk 5 above)
Comment (azeemshaikh38) [reply cmd/package_managers.go:5]: No clue. Worth discussing this though. Maybe start a new issue for this?

I don't want to add breaking changes in this PR though, so will leave it as is for now.

File: clients/localdir/repo.go (hunk 4 above)
Comment (azeemshaikh38) [reply clients/localdir/repo.go:84]: Let me know if there is any other tests I can run for this.

Comment (laurentsimon) [reply clients/localdir/repo.go:84]: should be fine. Can't think of anything else atm

**Pull request merged**